tests/unit/dom.ts

```diff
@@ -0,0 +1,241 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import * as dom from 'src/dom';
+
+let element: HTMLElement;
+
+registerSuite({
+    name: 'dom',
+
+    setup() {
+        element = document.createElement('span');
+        element.id = 'id';
+    },
+
+    teardown() {
+        element.parentElement.removeChild(element);
+    },
+
+    byId: {
+        'result found'() {
+            document.body.appendChild(element);
+            assert.strictEqual(dom.byId('id').id, 'id');
+        },
+
+        'result not found'() {
+            assert.isNull(dom.byId('undefined'))
+        }
+    },
+
+    place: {
+        'throws if reference node not in DOM'() {
+            let node = document.createElement('div'),
+                reference = document.createElement('h1');
+
+            assert.throws(function () {
+                dom.place(node, dom.Position.FirstIn, reference);
+            });
+        },
+
+        'relative element'() {
+            let node = document.createElement('div'),
+                reference = document.createElement('h1');
+
+            document.body.appendChild(reference);
+            dom.place(node, dom.Position.FirstIn, reference);
+            assert.equal(reference.firstElementChild.nodeName, node.nodeName);
+        },
+
+        'when third argument is Position.Before, the first arg is placed before the second'() {
```

Comment:
All of these descriptions that say "when third argument is" should say "when second argument is" - I assume this is another artifact of direct lifting from Dojo 1